tests/puzzle/test_solver.py

```diff
@@ -1989,71 +1989,37 @@ def test_solver_duplicate_dependencies_with_overlapping_markers_complex(
         ],
     )
     opencv_requires = {dep.to_pep_508() for dep in ops[-1].package.requires}
-    expectation = (
```

Comment:
Notably, I removed the distinction between "concise solution" and "current solution" because we have reached a solution which is even more concise than the "concise solution".